chain-abci/src/enclave_bridge/edp/tdbe.rs

```diff
@@ -0,0 +1,238 @@
+            enclave_builder.usercall_extension(self);
+
+            let enclave = enclave_builder
+                .build(&mut device)
+                .expect("Failed to build enclave");
+            enclave.run().expect("Failed to start enclave")
+        });
+    }
+}
+
+/// Fetches connection details of remote TDBE server using TM RPC
+fn fetch_remote_tdbe_connection_details(
+    _tdbe_config: &TdbeConfig,
+) -> std::io::Result<(Option<String>, Option<String>)> {
+    // TODO: Fetch connection details using TM RPC (use `client-common::TendermintRpcClient`?)
+    Ok((None, None))
+}
+
+fn spawn_chain_abci_thread(mut receiver: UnixStream, storage: Arc<dyn KeyValueDB>) {
+    let _ = thread::spawn(move || {
+        let storage = chain_storage::ReadOnlyStorage::new_db(storage);
+
+        while let Ok(enclave_request) = EnclaveRequest::read_from(&mut receiver) {
+            match enclave_request {
+                EnclaveRequest::GetSealedTxData { txids } => {
+                    let response =
+                        EnclaveResponse::GetSealedTxData(get_sealed_tx_data(txids, &storage));
+                    response
+                        .write_to(&mut receiver)
+                        .expect("Unable to write to TDBE <-> chain-abci unix stream");
+                }
+                _ => unreachable!("TDBE can only send `GetSealedTxData` request"),
+            }
+        }
+    });
+}
+
+fn get_sealed_tx_data(txids: Vec<TxId>, storage: &ReadOnlyStorage) -> Option<Vec<SealedLog>> {
+    let mut result = Vec::with_capacity(txids.len());
+
+    for txid in txids {
+        if let Some(txin) = storage.get_sealed_log(&txid) {
+            result.push(txin);
+        } else {
+            return None;
+        }
+    }
+
+    Some(result)
+}
+
+fn spawn_persistence_thread(mut receiver: UnixStream, storage: Arc<dyn KeyValueDB>) {
+    let _ = thread::spawn(move || {
+        let storage = chain_storage::Storage::new_db(storage);
+        let mut buffer = HashMap::new();
+        let mut kvdb = chain_storage::buffer::BufferStore::new(&storage, &mut buffer);
+
+        while let Ok(persistence_command) = PersistenceCommand::read_from(&mut receiver) {
+            chain_storage::store_sealed_log(
+                &mut kvdb,
+                &persistence_command.transaction_id,
+                &persistence_command.sealed_log,
+            );
+        }
+    });
+}
+
+#[allow(clippy::type_complexity)]
+impl UsercallExtension for TdbeApp {
+    fn connect_stream<'future>(
+        &'future self,
+        addr: &'future str,
+        _local_addr: Option<&'future mut String>,
+        _peer_addr: Option<&'future mut String>,
+    ) -> Pin<Box<dyn Future<Output = io::Result<Option<Box<dyn AsyncStream>>>> + 'future>> {
+        async fn connect_stream_inner(
+            this: &TdbeApp,
+            addr: &str,
+        ) -> io::Result<Option<Box<dyn AsyncStream>>> {
+            match addr {
+                "init" => {
```

Comment:
init is too broad, how about adding
"..._init"

Reply:
Added comments